galleries/examples/subplots_axes_and_figures/align_labels_demo.py

```diff
@@ -17,25 +17,19 @@
 
 fig, axs = plt.subplots(2, 2, layout='constrained')
 
-ax = axs[0][0]
-ax.plot(np.arange(0, 1e6, 1000))
-ax.set_title('Title0 0')
-ax.set_ylabel('YLabel0 0')
-
-ax = axs[0][1]
-ax.plot(np.arange(1., 0., -0.1) * 2000., np.arange(1., 0., -0.1))
-ax.set_title('Title0 1')
-ax.xaxis.tick_top()
-ax.tick_params(axis='x', rotation=55)
-
-
 for i in range(2):
-    ax = axs[1][i]
-    ax.plot(np.arange(1., 0., -0.1) * 2000., np.arange(1., 0., -0.1))
-    ax.set_ylabel('YLabel1 %d' % i)
-    ax.set_xlabel('XLabel1 %d' % i)
-    if i == 0:
-        ax.tick_params(axis='x', rotation=55)
+    for j in range(2):
+        ax = axs[i][j]
+        ax.plot(np.arange(1., 0., -0.1) * 1000., np.arange(1., 0., -0.1))
+        ax.set_title(f'Title {i} {j}')
+        ax.set_xlabel(f'XLabel {i} {j}')
+        ax.set_ylabel(f'YLabel {i} {j}')
+        if (i == 0 and j == 1) or (i == 1 and j == 0):
+            if i == 0 and j == 1:
+                ax.xaxis.tick_top()
+            ax.set_xticks(np.linspace(0, 1000, 5))
+            ax.set_xticklabels([250 * n for n in range(5)])
+            ax.xaxis.set_tick_params(rotation=55, rotation_mode='xtick')
```

Comment:
TYSM for explaining so clearly, and for your hard work to help me. I really appreciate it! 🩵

A few questions: 

There seems to be an issue with
```python
 ax.tick_params(axis='x', rotation=55, rotation_mode='xtick')
```
It leads to
```
AttributeError: Line2D.set() got an unexpected keyword argument 'ion_mode'
```
Tracing to
```python 
class Tick(martist.Artist):
                   ⋮
      def __init__(.**kwargs):
                   ⋮
            grid_kw = {k[5:]: v for k, v in kwargs.items()}
```

The fix I came up with is
```python 
grid_kw = {k[5:]: v for k, v in kwargs.items() if k != "rotation_mode"}
```
to filter out `rotation_mode` and stopping `ion_mode` from being passed to the `Line2D` object.

I'm not sure if it would be better to apply it separately, to the `__init__` method, though? 

```python 
ax.tick_params(axis='x', rotation=55, rotation_mode='xtick')
```
seems to apply `rotation_mode` to only the first tick on the axis. Is there a delayed creation of the ticks at render time? so the `rotation_mode` isn't being applied to all the labels? 

I've tried using

```python 
ax.set_xticks(ax.get_xticks())
ax.tick_params(axis='x', rotation=55, rotation_mode='xtick')
```
which seems to create the ticks before setting `tick_params`, but I'm not sure if it's a good solution?